lldb/bindings/python/python-wrapper.swig

```diff
@@ -642,6 +649,16 @@ bool lldb_private::python::SWIGBridge::LLDBSwigPythonCallCommandObject(
   pfunc(SWIGBridge::ToSWIGWrapper(std::move(debugger)), PythonString(args),
         SWIGBridge::ToSWIGWrapper(exe_ctx_ref_sp), cmd_retobj_arg.obj());
 
+  if (PyErr_Occurred()) {
+    py_err_cleaner.~PyErr_Cleaner();
```

Comment:
I'm pretty sure this is undefined behavior (you calling the destructor does not stop the compiler from calling it again).

I think that a proper fix for this would be to change `PythonCallable::operator()` to return `Expected<PythonObject>` (and convert the python exception to llvm::error, etc). A lot of the PythonDataObject methods were converted to do that (see comment at the top of the header), but it looks like this is not one of them.

Reply:
I replaced the explicit destructor call.

The change to `PythonCallable::operator()` sounds good, but looks like a wide ranging change. I'll set that aside for a future change.

Reply:
Fair enough. How about then making a new function (`Expected<PythonObject> PythonCallable::Call(/*something*/)` which does the right thing, and then porting callers you care about to the new API?